ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/viewmodel/CustomerCenterViewModel.kt

```diff
@@ -235,15 +235,20 @@ internal class CustomerCenterViewModelImpl(
         }
     }
 
-    override fun dismissRestoreDialog() {
-        _state.update { currentState ->
-            if (currentState is CustomerCenterState.Success) {
-                currentState.copy(
-                    showRestoreDialog = false,
-                    restorePurchasesState = RestorePurchasesState.INITIAL,
-                )
-            } else {
-                currentState
+    override suspend fun dismissRestoreDialog(restored: Boolean) {
+        if (restored) {
+            // Reload customer center data to refresh the UI with the latest information
+            loadCustomerCenter()
```

Comment:
Hmm does it make sense to just do this always after dismissing the restore dialog, even if there were no purchases? I don't think it would be a huge deal and would simplify the code but not sure if there are reasons for it.

Reply:
you know what? I like it. I wasn't a fan of this restored boolean but didn't find an alternative. If someone is opening the restore dialog anyways it would be good to reload just in case.